app/services/search/google_search_service.rb

```diff
@@ -4,18 +4,99 @@ module Search
   class GoogleSearchService
     GOOGLE_SEARCH_URL = 'https://www.google.com/search?q=%{keyword}&hl=%{language}&lr=%{language}'
 
+    USER_AGENT_HEADER = 'User-Agent'
+    USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36'
+
+    KEY_DIV_TOP_ADWORDS_CONTAINER = 'div.pla-unit-container'
+    KEY_DIV_NORMAL_ADWORDS_CONTAINER = 'div.v5yQqb'
+    KEY_DIV_NON_ADWORDS_CONTAINER = 'div.yuRUbf'
+
+    KEY_A_TAG = 'a'
+    KEY_A_TAG_ADWORDS_TOP_CLASS = 'a[data-impdclcc]'
+    KEY_A_TAG_HYPERLINK = 'a[href]'
+
+    KEY_HREF_ATTRIBUTE = 'href'
+
     def initialize(keyword, language = 'en')
       @search_url = format(GOOGLE_SEARCH_URL, keyword: keyword, language: language)
     end
 
     def search
       uri = URI.parse(@search_url)
 
-      response = Faraday.get(uri)
+      response = Faraday.get(uri, nil, USER_AGENT_HEADER => USER_AGENT)
```

Comment:
We might need to random the user agent, as Google will detect spam when we use the same one.

Reply:
@hoangmirs I noticed that the search page is returning different results since we are generating UA randomly for usage. This is the challenge to achieve the parsing logic to make it works for all cases. Do you have any advice to tackle this? 🙏

Reply:
Nevermind 😄 I changed the UA generator to generate specific UA for chrome browser so it works